Added deployment section in the landing page (#155)
strimzi/strimzi-canary: README.md

It acts as an indicator of whether Kafka clusters are operating correctly.
This is achieved by creating a canary topic and periodically producing and consuming events on the topic and getting metrics out of these exchanges.

## Deployment

In order to deploy the Strimzi canary in the Kubernetes cluster alongside the already running Apache Kafka cluster, first download the installation files from one of the [releases](https://github.com/strimzi/strimzi-canary/releases) you are interested in.

> Unzip the release package and edit the `Deployment` in order to specify the right bootstrap servers for connecting to the Apache Kafka cluster; it is defined by the `KAFKA_BOOTSTRAP_SERVERS` environment variable.

Author: I am not sure I really like this shape. It sounds more a product documentation, I would like to have a more "conversational" way to explain how to deploy. Not sure what I am saying makes sense to you.

> Unzip the release package and edit the `Deployment` in order to specify the right bootstrap servers for connecting to the Apache Kafka cluster; it is defined by the `KAFKA_BOOTSTRAP_SERVERS` environment variable.

Reviewer: Makes sense. I removed the numbered list. Does that help?

> Unzip the release package and edit the `Deployment` in order to specify the right bootstrap servers for connecting to the Apache Kafka cluster; it is defined by the `KAFKA_BOOTSTRAP_SERVERS` environment variable.

Author: Maybe yes :-) Thanks

The provided `Deployment` has also TLS disabled by default (`TLS_ENABLED` as `false`) and the canary producing and consuming messages every 10 seconds (`RECONCILE_INTERVAL_MS` as `"10000"`).

Then just run the following command:

```shell
kubectl apply -f ./install
```

Other than creating the corresponding `Deployment`, the canary will run with a specific `ServiceAccount` and a `Service` will be created in order to make the Prometheus metrics accessible via HTTP on port `8080`.

### Encryption and TLS

> If the Apache Kafka cluster has the TLS protocol enabled on the listener where the canary should connect to in order to encrypt the traffic, enable the TLS configuration in the canary `Deployment` using the corresponding environment variables.

Reviewer: ```suggestion
If your Apache Kafka cluster has TLS enabled to encrypt traffic on the listener canary will use to connect, enable TLS for the canary tool as well.
```

> The `TLS_ENABLED` has to be set as `true` and the `TLS_CA_CERT` has to contain the cluster CA certificate, in PEM format, used to sign the broker certificates; if left empty, the canary will use the system certificates.
> Leaving the Strimzi Cluster Operator genberating the cluster CA certificate, it can be extracted from the corresponding `<cluster_name>-cluster-ca-cert` `Secret`.

Reviewer: ```suggestion
Set `TLS_ENABLED` to `true` in the `Deployment` file. You'll also need a cluster CA certificate in PEM format that validates the identity of the Kafka brokers. You can reference a cluster CA certificate using the  `TLS_CA_CERT` environment variable. If you leave `TLS_CA_CERT` empty, canary can use the system certificates generated by the Cluster Operator. You can extract the cluster CA certificate from the `<cluster_name>-cluster-ca-cert` `Secret`.
```

Reviewer: Might not have read the original draft correctly, but I'm assuming you have to reference a certificate whether it's your own or the system cert

Author: No it's not this way. You can specify the cluster CA that could be generated by Strimzi or the one provided by the user (but anyway I think these are too many details). If you leave the env var empty, you are relying on "system certificates" that are the ones for well known CAs and are typically already installed on any laptop we use (see for example Verisign, Let's Encrypt and so on).

Author: I will change it from your suggestion.

### Authentication and authorization

> If the Apache Kafka cluster has authentication enabled by using TLS mutual (client) authentication, the canary has to be configured with the proper certificate and private key, in PEM format, by using the corresponding environment variables `TLS_CLIENT_CERT` and `TLS_CLIENT_KEY`.

Reviewer: ```suggestion
If the Apache Kafka cluster has TLS mutual (client) authentication enabled, the canary has to be configured with a client certificate and private key in PEM format. Use the corresponding environment variables `TLS_CLIENT_CERT` and `TLS_CLIENT_KEY`.
```

> If using the Strimzi User Operator, those data are provided by the corresponding `Secret` for the `KafkaUser` configured with TLS authentication.

Reviewer: ```suggestion
If you're using the Strimzi User Operator, the values for these environment variables are provided by the `Secret` for the `KafkaUser` configured with TLS authentication.
```

> If the Apache Kafka cluster has authentication enabled with one of the supported SASL mechanisms as `PLAIN`, `SCRAM-SHA-256` and `SCRAM-SHA-512`, the canary has to be configured to use it.

Reviewer: ```suggestion
If the Apache Kafka cluster has authentication enabled with the `PLAIN`, `SCRAM-SHA-256`, or `SCRAM-SHA-512` SASL mechanism, the canary must be configured to use it as well.
```

> The SASL mechanism has to be defined using the `SASL_MECHANISM` environment variable while username and password by using `SASL_USER` and `SASL_PASSWORD`.

Reviewer: ```suggestion
The SASL mechanism is specified using the `SASL_MECHANISM` environment variable. The username and password are specified using the `SASL_USER` and `SASL_PASSWORD` environment variables.
```

> If using the Strimzi User Operator, those data are provided by the corresponding `Secret` for the `KafkaUser` configured to use one of the SASL authentication mechanisms.

Reviewer: ```suggestion
If you're using the Strimzi User Operator, the values for these environment variables are provided by the corresponding `Secret` for the `KafkaUser` configured to use one of the SASL authentication mechanisms.
```

Reviewer: I wonder if it's worth showing an example config with all the potential authentication settings?

Author: We thought that it's not necessary for now. It's enough describing the deployment.

It acts as an indicator of whether Kafka clusters are operating correctly.
This is achieved by creating a canary topic and periodically producing and consuming events on the topic and getting metrics out of these exchanges.

## Deployment

In order to deploy the Strimzi canary in the Kubernetes cluster alongside the already running Apache Kafka cluster, first download the installation files from one of the [releases](https://github.com/strimzi/strimzi-canary/releases) you are interested in.
Unzip the release package and edit the `Deployment` in order to specify the right bootstrap servers for connecting to the Apache Kafka cluster; it is defined by the `KAFKA_BOOTSTRAP_SERVERS` environment variable.
The provided `Deployment` has also TLS disabled by default (`TLS_ENABLED` as `false`) and the canary producing and consuming messages every 10 seconds (`RECONCILE_INTERVAL_MS` as `"10000"`).

Then just run the following command:

```shell
kubectl apply -f ./install
```

Other than creating the corresponding `Deployment`, the canary will run with a specific `ServiceAccount` and a `Service` will be created in order to make the Prometheus metrics accessible via HTTP on port `8080`.

> ### Encryption and TLS authentication

Reviewer 2: Should this maybe be split into two sections? One for encryption and one for TLS client auth? It would make it more clear that you can also use encryption with SASL.

Author: @scholzj agree. I just changed by having an "Encryption and TLS" section (just for TLS) separated from the "Authentication and authorization" with TLS client auth and SASL auth.
